apps/src/code-studio/pd/application/teacher1819/Section4SummerWorkshop.jsx

```diff
@@ -0,0 +1,356 @@
+        <h3>Section 4: {SectionHeaders.section4SummerWorkshop}</h3>
+
+        <p>
+          All participants in Code.org’s Professional Learning Program are required to attend a five-day
+          in-person summer workshop. These workshops are either hosted locally by Code.org’s Regional Partners,
+          or by Code.org at Code.org’s TeacherCons. Participants are assigned to workshops based on their
+          Regional Partner. Flights, lodging, and meals will be provided for TeacherCon attendees; meals
+          (and in some cases travel costs) will be provided for summer workshops hosted by Regional Partners.
+        </p>
+
+        {this.renderAssignedWorkshopList()}
+
+        {this.isUnableToAttendAssignedWorkshop() && [1,2].includes(this.props.data.regionalPartnerGroup) &&
+          <div style={styles.indented}>
+            <p style={styles.formText}>
+              <strong>
+                We strongly encourage participants to attend their assigned summer workshop (based on the region
+                in which you currently teach), so that you can meet the other teachers, facilitators, and
+                Regional Partners with whom you will work in 2018-19.
+              </strong>
+            </p>
+
+            {this.renderAlternateWorkshopList()}
+          </div>
+        }
+
+        {this.props.data.regionalPartnerGroup === 1 &&
+          <div>
+            <label>
+              There may be a fee associated with your summer workshop.
+              Please carefully
+              {' '}<a href={WORKSHOP_FEES_URL} target="_blank">check this list</a>{' '}
+              to find more information about your workshop.
+            </label>
+
+            {this.singleCheckboxFor("understandFee")}
+            {this.radioButtonsFor("payFee")}
+
+            {this.props.data.payFee === NO_PAY_FEE &&
+              this.singleCheckboxFor("considerForFunding", {
+                required: false,
+                style: styles.checkBoxAfterButtonList
+              })
+            }
+          </div>
+        }
+
+        <br />
+        <div style={styles.formText}>
+          <strong>
+            As a reminder, teachers in this program are required to participate in both:
+            <ul>
+              <li>
+                One five-day, in-person summer workshop in 2018
+              </li>
+              <li>
+                Four one-day, in-person local workshops during the 2018 - 19 school year (typically held on Saturdays)
+              </li>
+            </ul>
+          </strong>
+        </div>
+
+        {this.radioButtonsWithAdditionalTextFieldsFor("committed", {
+          [NO_EXPLAIN] : 'explain'
+        })}
+
+        {this.radioButtonsFor("willingToTravel")}
+
+      </FormGroup>
+    );
+  }
+
+  /**
+   * @override
+   */
+  static getDynamicallyRequiredFields(data) {
+    const requiredFields = [];
+
+    if (data.regionalPartnerGroup === 3 || data.regionalPartnerWorkshopCount === 1) {
```

Comment:
Might as well use the same .includes syntax that you are using below

Reply:
Can't. These are different values: the group and the workshop count. 

We have these states:
partner in group 1-2: use workshops (or none)
partner in group 3: use workshops if they exist, otherwise TeacherCon